src/log4net/Core/LoggingEvent.cs

```diff
@@ -826,11 +825,10 @@ private static string TryReadWindowsIdentityUserName()
         {
           try
           {
-            if (System.Threading.Thread.CurrentPrincipal is not null &&
-                System.Threading.Thread.CurrentPrincipal.Identity is not null &&
-                System.Threading.Thread.CurrentPrincipal.Identity.Name is not null)
+            if (Thread.CurrentPrincipal is not null &&
```

Comment:
I think we can shorten this even more:
```suggestion
            if (Thread.CurrentPrincipal?.Identity?.Name is string name)
```